backend/search/models.py

```diff
@@ -474,5 +474,36 @@ def augment_with_variables(self, matches: ResultSet) -> ResultSet:
             m.variables = f'<vars>{vars_str}</vars>'
         return matches
 
+    def augment_with_context(self, matches: ResultSet, limit: int) -> ResultSet:
+        """Fetch preceding and following sentences for matches in the result set,
+        up to limit results"""
+        i = 0
+        for match in matches:
+            if i > limit:
+                break
+
+            sentid = match._match.sentid
+            if '+match' in sentid:
```

Comment:
I will make an issue to split the sentence id and match number because this will lead to bugs at some point. For now, I would suggest changing it to what the frontend does, with a regexp: `hitId.replace(/\+match=\d+$/, '')`